commit f541ff89c2d66197e371ed6f85174bd732afd8db
Author: Ben Cheng <bccheng@android.com>
Date:   Tue Apr 2 19:31:03 2013 +0800

    [4.8] Enable armv7/thumb2 multilib for arm-eabi.
    
    Configured multilib:
    
    > arm-eabi-gcc -print-multi-lib
    .;
    thumb;@mthumb
    thumb/thumb2;@mthumb@march=armv7
    
    See 978fb74a5c086cc5572dc2e3f37d207acad969d7
    
    Change-Id: Ia04c9e23cb5abb924702639c6c0f3afc9c485ff0

diff --git gcc-4.8/gcc/config/arm/t-arm-elf gcc-4.8/gcc/config/arm/t-arm-elf
index 60747d3..76b0230 100644
--- gcc-4.8/gcc/config/arm/t-arm-elf
+++ gcc-4.8/gcc/config/arm/t-arm-elf
@@ -25,22 +25,27 @@ MULTILIB_MATCHES     =
 #MULTILIB_DIRNAMES    += fa526 fa626 fa606te fa626te fmp626 fa726te
 #MULTILIB_EXCEPTIONS  += *mthumb*/*mcpu=fa526 *mthumb*/*mcpu=fa626
 
-#MULTILIB_OPTIONS      += march=armv7
-#MULTILIB_DIRNAMES     += thumb2
-#MULTILIB_EXCEPTIONS   += march=armv7* marm/*march=armv7*
-#MULTILIB_MATCHES      += march?armv7=march?armv7-a
-#MULTILIB_MATCHES      += march?armv7=march?armv7-r
-#MULTILIB_MATCHES      += march?armv7=march?armv7-m
-#MULTILIB_MATCHES      += march?armv7=mcpu?cortex-a8
-#MULTILIB_MATCHES      += march?armv7=mcpu?cortex-r4
-#MULTILIB_MATCHES      += march?armv7=mcpu?cortex-m3
+# Enable these for ANDROID arm-eabi toolchain
+MULTILIB_OPTIONS      += march=armv7
+MULTILIB_DIRNAMES     += thumb2
+MULTILIB_EXCEPTIONS   += march=armv7* marm/*march=armv7*
+MULTILIB_MATCHES      += march?armv7=march?armv7-a
+MULTILIB_MATCHES      += march?armv7=march?armv7-r
+MULTILIB_MATCHES      += march?armv7=march?armv7-m
+MULTILIB_MATCHES      += march?armv7=mcpu?cortex-a8
+MULTILIB_MATCHES      += march?armv7=mcpu?cortex-r4
+MULTILIB_MATCHES      += march?armv7=mcpu?cortex-m3
+MULTILIB_MATCHES      += march?armv7=mcpu?cortex-m4
 
 # Not quite true.  We can support hard-vfp calling in Thumb2, but how do we
 # express that here?  Also, we really need architecture v5e or later
 # (mcrr etc).
-MULTILIB_OPTIONS       += mfloat-abi=hard
-MULTILIB_DIRNAMES      += fpu
-MULTILIB_EXCEPTIONS    += *mthumb/*mfloat-abi=hard*
+
+# Disable these for ANDROID arm-eabi toolchain
+#MULTILIB_OPTIONS       += mfloat-abi=hard
+#MULTILIB_DIRNAMES      += fpu
+#MULTILIB_EXCEPTIONS    += *mthumb/*mfloat-abi=hard*
+
 #MULTILIB_EXCEPTIONS    += *mcpu=fa526/*mfloat-abi=hard*
 #MULTILIB_EXCEPTIONS    += *mcpu=fa626/*mfloat-abi=hard*
 
